petab/core.py

```diff
@@ -507,10 +512,90 @@ def _apply_overrides(
                 overrides, condition_ix,
                 row.observableId, override_type='noise')
 
+    fill_in_nominal_values(mapping, parameter_df)
+
     handle_missing_overrides(mapping, warn=warn_unmapped)
     return mapping
 
 
+def _apply_dynamic_parameter_overrides(mapping,
+                                       condition_df: pd.DataFrame,
+                                       parameter_df: pd.DataFrame,
+                                       par_sim_id_to_ix):
+    """Apply dynamic parameter overrides from condition table (inplace).
+
+    Arguments:
+        mapping, par_sim_id_to_ix:
+            see get_optimization_to_simulation_parameter_mapping
+        condition_df, parameter_df: PEtab condition and parameter table
+    """
+    for overridee_id in condition_df.columns:
+        if overridee_id == 'conditionName':
+            continue
+        if condition_df[overridee_id].dtype != 'O':
+            continue
+
+        for condition_idx, overrider_id \
```

Comment:
can we be sure that the condition_idx will be the same one as the one generated via get_simulation_conditions? where also preeq is considered.

Reply:
We don't change any ordering here and don't try find the condition anywhere else. We are just walking over the array. I don't think anything can go wrong there.